app/controllers/order_items_controller.rb

```diff
@@ -1,17 +1,37 @@
 class OrderItemsController < ApplicationController
 
+  before_action :require_login, only: [:index]
+
   def index
-    @merchant_order_items = OrderItem.where(merchant_id: params[:merchant_id])
+    if params[:merchant_id].to_i != @login_merchant.id
+      flash[:warning] = "You are not authorized to view this section"
+      redirect_back(fallback_location: root_path)
+      return
+    end
 
-    if @merchant_order_items == nil
-      flash[:error] = "Merchant does not have order items in our database."
-      head :not_found
+    # Adding Query String for filtered views: https://stackoverflow.com/questions/2695538/add-querystring-parameters-to-link-to
+    @status = params["status"]
+    if @status == "pending"
+      @merchant_order_items = OrderItem.where(merchant_id: params[:merchant_id], status: "pending")
+      return
+    elsif @status == "shipped"
+      @merchant_order_items = OrderItem.where(merchant_id: params[:merchant_id], status: "shipped")
+      return
+    else
+      @status = "all"
+      @merchant_order_items = OrderItem.where(merchant_id: params[:merchant_id])
+      return
     end
```

Comment:
Getting a merchant's order_items seems business logic-y to me -- I'd suggest putting that logic into a merchant model method (maybe with an opt argument for status so you can use it on lines 15, 18, and 22)